src/kademlia/dht_storage.cpp

```diff
@@ -532,6 +556,36 @@ namespace
 			if (!peers.empty() && peers.capacity() / peers.size() >= 4u)
 				peers.shrink_to_fit();
 		}
+
+		void refresh_infohashes_sample()
```

Comment:
thanks, now I understand, well, the problem is the exact nature of `T`, in this particular case is a pointer, but assuming `T` has `data()` and `size()` I think is doable, I want to give it a try in another PR to use it here